### PR TITLE
Upgrade ethereum-cryptography to 1.0 

### LINKED ISSUE #5122: Upgrade ethereum-cryptography to 1.0
You're using 0.2 in util. 1.0 should be used, because it consumes 5 dependencies instead of 38; and has massive security improvement.

Blog post: https://medium.com/nomic-foundation-blog/a-safer-smaller-and-faster-ethereum-cryptography-stack-5eeb47f62d79

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Fixes #5122

## Type of change

- [x] Upgrades

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-eth-accounts/test/fixtures/account.ts [hunk 1]
```diff
@@ -244,54 +282,54 @@ export const invalidEncryptData: [
...
-			{
-				iv: 'bfb43120ae00e9de110f8325143a2709',
-				salt: '210d0ec956787d865358ac45716e6dd42e68d48e346d795746509523aeb477dd',
-				c: 100,
-				kdf: 'pbkdf2',
-			},
-		],
-		new PBKDF2IterationsError(),
-	],
+	// [
```
Comment (nikoulai) [inline packages/web3-eth-accounts/test/fixtures/account.ts]: Why we need to comment these ?

**Review by nikoulai — APPROVED** [review]

File: packages/web3-eth-accounts/src/account.ts [hunk 2]
```diff
@@ -103,31 +134,27 @@ export const hashMessage = (message: string): string => {
...
 
 	const hash = hashMessage(data);
 
-	const signObj = ecdsaSign(
-		Buffer.from(hash.substring(2), 'hex'),
-		Buffer.from(privateKeyParam, 'hex'),
-	);
+	const [signature, recoverId] = signSync(hash.substring(2), privateKeyBuffer, {
+		recovered: true,
+		der: false,
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/account.ts]:
```suggestion
		recovered: true, // makes signatures compatible with libsecp256k1
		der: false, //returned signature should be in DER format ( non compact )
```

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: packages/web3-eth-accounts/test/fixtures/account.ts [hunk 3]
```diff
@@ -392,3 +430,16 @@ export const invalidDecryptData: [[any, string], InvalidKdfError | KeyDerivation
...
+	[
+		'Some long text with integers 1233 and special characters and unicode \u1234 as well.',
+		'0xff21294f27c6b1e416215feb0b0b904c552c874c4e11b2314dd3afc1714ed8a8',
+	],
+	[
+		'Some long text with integers 1233 and special characters and unicode \u1234 as well.',
+		'0xff21294f27c6b1e416215feb0b0b904c552c874c4e11b2314dd3afc1714ed8a8',
+	],
+	['non utf8 string', '0x8862c6a425a83c082216090e4f0e03b64106189e93c29b11d0112e77b477cce2'],
+];
```
Comment (jdevcs) [inline packages/web3-eth-accounts/test/fixtures/account.ts:442]:
```suggestion
['','0x5f35dce98ba4fba25530a026ed80b2cecdaa31091ba4958b99b52ea1d068adad']
];
```

good to add edge case

File: packages/web3-eth-accounts/test/fixtures/account.ts [hunk 4]
```diff
@@ -392,3 +430,16 @@ export const invalidDecryptData: [[any, string], InvalidKdfError | KeyDerivation
...
+export const validHashMessageData: [string, string][] = [
+	['🤗', '0x716ce69c5d2d629c168bc02e24a961456bdc5a362d366119305aea73978a0332'],
+	[
+		'Some long text with integers 1233 and special characters and unicode \u1234 as well.',
+		'0xff21294f27c6b1e416215feb0b0b904c552c874c4e11b2314dd3afc1714ed8a8',
+	],
+	[
+		'Some long text with integers 1233 and special characters and unicode \u1234 as well.',
+		'0xff21294f27c6b1e416215feb0b0b904c552c874c4e11b2314dd3afc1714ed8a8',
+	],
```
Comment (jdevcs) [inline packages/web3-eth-accounts/test/fixtures/account.ts]: remove duplicate test data here

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: packages/web3-eth-accounts/src/account.ts [hunk 5]
```diff
@@ -77,11 +105,14 @@ export const hashMessage = (message: string): string => {
...
+	const preamble = Buffer.from(
+		`\x19Ethereum Signed Message:\n${messageBytes.byteLength}`,
+		'utf8',
+	);
 
-	const ethMessage = Buffer.concat([Buffer.from(preamble), Buffer.from(messageBytes)]);
+	const ethMessage = Buffer.concat([preamble, messageBytes]);
 
-	return `0x${Buffer.from(keccak256(ethMessage)).toString('hex')}`;
+	return sha3Raw(ethMessage);
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/account.ts]:
```suggestion
	return sha3Raw(ethMessage);  // using keccak in web3-utils.sha3Raw instead of SHA3 (NIST Standard) as both are different
```

File: packages/web3-eth-accounts/src/account.ts [hunk 6]
```diff
@@ -234,7 +261,7 @@ export const signTransaction = async (
 	}
 
 	const rawTx = bytesToHex(signedTx.serialize());
-	const txHash = keccak256(hexToBytes(rawTx));
+	const txHash = sha3Raw(rawTx);
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/account.ts]:
```suggestion
	const txHash = sha3Raw(rawTx); // using keccak in web3-utils.sha3Raw instead of SHA3 (NIST Standard) as both are different
```

File: packages/web3-eth-accounts/test/fixtures/account.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/web3-eth-accounts/test/fixtures/account.ts]: Forget to uncomment. Fixed it.

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]
